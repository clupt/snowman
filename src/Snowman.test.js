import Snowman from "./Snowman";
import React from "react";
import { render, debug, fireEvent } from "@testing-library/react";
import TEST_IMAGES from "./_testCommon";

describe("tests for reaching max incorrect guesses", function () {
  function createContainer() {
    const { container } = render (
      <Snowman
        images={TEST_IMAGES}
        words={['apple']}
        maxWrong={2}
      />
    );
    return container;
  };

  it("stops showing button area after nWrong reaches maxGuesses", function () {
    const container = createContainer();
    expect(container.querySelector('.Snowman-buttons')).toBeInTheDocument();
    //fire off 2 btns
    const xBtn = container.querySelector('button[value="x"]');
    const yBtn = container.querySelector('button[value="y"]');
    console.log("xBtn=", xBtn, "yBtn=", yBtn);
    fireEvent.click(xBtn);
    fireEvent.click(yBtn);
    expect(container.querySelector('.Snowman-buttons')).not.toBeInTheDocument();
  });

  it("matches snapshot", function () {
    const container = createContainer();
    const xBtn = container.querySelector('button[value="x"]');
    const yBtn = container.querySelector('button[value="y"]');
    fireEvent.click(xBtn);
    fireEvent.click(yBtn);
    expect(container).toMatchSnapshot();
  });
});
