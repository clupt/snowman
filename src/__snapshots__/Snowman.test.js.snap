// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for reaching max incorrect guesses matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="2"
      src="[object Object]"
    />
    <p
      class="Snowman-nWrong"
    >
      Number wrong: 
      2
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
  </div>
</div>
`;
